### PR TITLE
Adapt integration tests to handle possibly orphaned resources

### LINKED ISSUE #498: Improve CI to prevent any orphaned resources
**What would you like to be added**:
Improve the CI script to prevent any orphaned resources

**Why is this needed**:
We have in the past observed orphaned volumes on the deletion of machines due to new API changes. Shall regressions must be checked by double-checking on infra resources (VMs. Disks, NICs) before CI tests terminate.

### PR DESCRIPTION
**Pull request by hardikdr**
**What this PR does / why we need it**: Adapts the integration test script to fail if there are any leftover orphan resources after the tests.

**Which issue(s) this PR fixes**:
Fixes #498 

**Special notes for your reviewer**:

**Release note**:

```improvement operator
Adapted integration tests to handle possibly orphaned resources.
```

### CODE REVIEW

File: .ci/integration-tests [hunk 1]
```diff
@@ -224,6 +261,37 @@ function hf_check_ms_freeze() {
     fi
 }
 
+function hf_orphan_resources() {
+    if [ $1 == "aws" ]; then
+        # Check if there are any VMs matching the tag exists.
```
Comment (prashanth26) [inline .ci/integration-tests:266]: Does it make sense to check for orphan NICs also here?

File: .ci/integration-tests [hunk 2]
```diff
@@ -288,6 +356,19 @@ function tc_machine_deployment() {
     printf "\nCompleted TestCase\n"
 }
 
+function tc_check_orphan_resources() {
+    
+    printf "\nStarting TestCase: %s" "${FUNCNAME[0]}"
+     
+    # Check if there are any leftover resources.
+    if ! hf_orphan_resources "$provider"; then
+        printf "\n\t Failed: There are orphan resources in the cloud. Exiting the test."
```
Comment (prashanth26) [inline .ci/integration-tests:365]: Also, should we delete these orphan resources if/when we find them?

Comment (hardikdr) [reply .ci/integration-tests:365]: Sure, we can do that, I'll adapt it.

Comment (hardikdr) [reply .ci/integration-tests:402]: done - https://github.com/gardener/machine-controller-manager/pull/550/commits/5c9b6d9bfbe1a621e7c459774efa8f830f280adf

File: .ci/integration-tests (hunk 1 above)
Comment (hardikdr) [reply .ci/integration-tests:268]: As discussed skipping the nic part.

**Review by prashanth26 — APPROVED** [review]

**Pull request merged**